Citizen Lab consent form

rpanah opened the pull request:
@ben-jones, please review (only the form contents were modified, but you should still check tags etc.).

ben-jones commented:
Does this still satisfy IRB requirements? We can't change the form without IRB approval for a change to the protocol

phillipa commented:
Stony Brook determined we are not human subjects research so we are switching to Toronto's form as we still have IRB approval there.

ben-jones commented:
Cool. I will look over this in a bit more depth. The first thing that comes up after downloading and rendering is that people have to select which country they want to do informed consent for and submit the consent without changing their country. @rpanah, can you remind me if we autopopulate the country? If not, we should either autopopulate the country or don't show the informed consent submit button until they select a country so that we force users to see content relevant to their country before consenting.

rpanah commented:
Is populated the same way as before.
On Mar 27, 2015 10:09 AM, "Ben Jones" <notifications@github.com> wrote:

> Cool. I will look over this in a bit more depth. The first thing that
> comes up after downloading and rendering is that people have to select
> which country they want to do informed consent for and submit the consent
> without changing their country. @rpanah <https://github.com/rpanah>, can
> you remind me if we autopopulate the country? If not, we should either
> autopopulate the country or don't show the informed consent submit button
> until they select a country so that we force users to see content relevant
> to their country before consenting.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/iclab/centinel-server/pull/65#issuecomment-86951303>.
>

ben-jones commented on static/informed_consent.html:
```diff
@@ -1,419 +1,413 @@
+          Experiment data may include issuing requests for URLs found in the URL list (similar to what a
+          user does manually with a Web browser, such as Firefox, but automated), DNS queries (mapping
+          hostnames such as Google.com to IP address such as 72.14.52.3), or traceroutes (measuring
+          network paths). Experiments may log network traffic associated with the experiment (e.g.,network packets and timestamps). The software will also contain a unique user ID to allow us to
```
can we do a line wrap to 80 characters here please? The command is M-q in emacs

ben-jones commented on static/informed_consent.html:
```diff
@@ -1,419 +1,413 @@
+            </div>
+
+            <div class="list-group-item">
+              <p>I will be given I can be provided with a copy of this signed and dated consent form.</p>
```
I think this is a typo. Was the typo part of the accepted protocol and can we change it?

rpanah replied:
It was part of it (I just copied text over). It should be fine to just correct it.

ben-jones commented on static/initial_informed_consent.html:
```diff
@@ -1,182 +1,268 @@
+            <div class="list-group-item">
+              <p>
+              Economist "Democracy Index score"<br/>
+              http://www.eiu.com/public/thankyou_download.aspx?activity=download&campaignid=DemocracyIndex12
```
What is the purpose of these links and can we remove them to ensure better privacy? The reason that we download all of the content to our server, then display what we downloaded is that we want to make sure that we aren't sending our users to the original source. If the user goes to the original source, then Freedom House or anyone else can have a list of our clients and we don't want that. Additionally, even if these are left in just for reference, there is still the potential that these links could identify the user due to browser prefetching (the browsing looking up the domains even though they are never visited)

rpanah replied:
While I agree with you, I'm not sure if we can remove them. I'll have to ask Citizen Lab folks about that.

rpanah commented:
@ben-jones, 
I have addressed all of the issues you mentioned (removed URLs from the form, fixed HTML formatting, corrected typo), and also fixed the bugs in all of your regular expressions (multiline wasn't enabled, so some instances didn't match) and enhanced them.
I also added a bit that removes the form tags (makes buttons and stuff go away, making it cleaner).
Please review again.
